src/System.Data.SqlClient/src/System/Data/Locale/LocaleMapper.Unix.cs

```diff
@@ -0,0 +1,482 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace System.Data
+{
+    /// <summary>
+    /// This a locale mapper for non windows
+    /// </summary>
+    internal class LocaleMapper
+    {
+        private static readonly Dictionary<int, LocaleCodePage> _mapper;
+        public static string LcidToLocaleNameInternal(int lcid)
+        {
+            return _mapper[lcid].LocaleName;
+        }
+
+        public static int LocaleNameToAnsiCodePage(string localeName)
+        {
+            return _mapper.FirstOrDefault(t=>t.Value.LocaleName == localeName).Value.CodePage;
+        }
+
+        public static int GetLcidForLocaleName(string localeName)
+        {
+            return _mapper.FirstOrDefault(t => t.Value.LocaleName == localeName).Key;
+        }
+
+        static LocaleMapper()
+        {
+            _mapper = new Dictionary<int, LocaleCodePage>
+            {
+                #region <<Locale Mapper>>
+                {1,new LocaleCodePage("ar",1256)},
```

Comment:
Nit: Can you apply the formatter for this map? 
The current layout looks very dense.